plans/init.pp

```diff
@@ -165,6 +173,9 @@
     $reboot_message_group = pick($reboot_message,
                                   $group_vars['patching_reboot_message'],
                                   'NOTICE: This system is currently being updated.')
+    $update_provider_group = pick($update_provider,
```

Comment:
```suggestion
    $update_provider_group = pick_default($update_provider,
```
Since your last option is `undef` i think  you need to use the `pick_default()` function,  because `pick()` will throw an error if all options are `undef`, see: https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/REFERENCE.md#pick_default